subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractDependencyResolutionTest.groovy

```diff
@@ -32,4 +34,9 @@ abstract class AbstractDependencyResolutionTest extends AbstractIntegrationSpec
     MavenFileRepository mavenRepo(String name = "repo") {
         return maven(name)
     }
+
+    protected ExecutionResult succeeds(String... tasks) {
+        executer.noDeprecationChecks()
```

Comment:
Because lots of dependency management tests have some assertions like:
```
file('build').hasDescendants('foo.jar','bar.jar')
```
and deprecation report would break them. Now I adjusted `hasDescendants()` to fix these failed tests.

Reply:
We don't need to change how hasDescendants works. We can output the warning saying "there were 3 deprecations" without writing the report (using an internal property). That way we still fail if there are deprecations without changing the output of the tests.

Reply:
We do have the internal property to control the report generation, however, I prefer to change `hasDescendants` because not only this test failed with `hasDescendants`. Exclude report in `hasDescendants` would help decrease the size of this PR, I think.